Compatibility matrix strategy (#313)
erlef/setup-beam: README.md

      - run: rebar3 ct
```

### Compatibility

> One can run matrix-style testing but with pinned compatible versions in the following manner:

Reviewer: ```suggestion
Matrix-style testing can be executed using pinned, compatible dependency versions by configuring the test matrix accordingly:
```

```yaml
  test:
    runs-on: ${{matrix.erlang.os}}
    name: Erlang/OTP ${{matrix.erlang.otp}} / rebar3 ${{matrix.erlang.rebar3}}
    strategy:
      matrix:

>         erlang:

Reviewer: This list will very easily get outdated. To ease management (and reduce the size) of the README, could you remove everything below 26?

          - otp: "21.3.8.17"
            rebar3: "3.15.2"
            os: ubuntu-20.04
          - otp: "22.3.4.9"
            rebar3: "3.16.1"
            os: ubuntu-20.04
          - otp: "23.3.4.5"
            rebar3: "3.16.1"
            os: ubuntu-22.04
          - otp: "24.3.4.17"
            rebar3: "3.16.1"
            os: ubuntu-22.04
          - otp: "25.3.2.15"
            rebar3: "3.22.1"
            os: ubuntu-24.04
          - otp: "26.2.5.5"
            rebar3: "3.22.1"
            os: ubuntu-24.04
          - otp: "27.1.2"
            rebar3: "3.22.1"
            os: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: erlef/setup-beam@v1

>         with:

Reviewer: If you really wanna make sure they're pinned you should use `version-type` `strict` in the example. And maybe could we reduce it a bit? Or is there a specific goal to having 21 through 27 (?) I'd be Ok with 25, 26, 27, atm.

Reviewer 2: <= 23 seems not to be available anymore.

Reviewer: > although currently the OTP versions seems to be too specific to be available

I don't get what this means.

Reviewer 2: Sorry. Got some "unavailable" errors in CI with the specific OTP versions from the example. Relaxing them to `major.0` solved it; although that makes the builds less deterministic.

Reviewer: I'm missing context here: what are you trying to achieve? The example states how one can use a strategy matrix, but that's not really `setup-beam`'s responsibility (in teaching, that is, since GitHub Actions is pretty well documented in that sense). On the other hand, if we're trying to show how one can achieve deterministic Elixir v. Erlang versions per OTP and OS, one way to do it (maybe the best way to do it) is to use `strict` and `version-type` option's value.

Reviewer 2: **Trying to:**
I am simply trying to achieve a pipeline which supports OTP 24 -> latest(-rc) and Elixir 1.16 -> latest. This requires some guidance as not all version combinations will work. So the example was a great help. However, I had to relax the versions a bit for it to work as some minor or patch versions were not available.

**but that's not really setup-beam's responsibility**
True. But it would be nice if I could somehow know which versions are available to pick from. Might have missed it, but I do not find a list for all available OTP versions.

It's the burden of examples: they get stale soon. Providing a little guidance for the reader how to resolve that issue might be all that's needed even though it's not a responsibility.

      - run: rebar3 ct
```

### Compatibility

One can run matrix-style testing but with pinned compatible versions in the following manner:

```yaml
  test:
    runs-on: ${{matrix.erlang.os}}
    name: Erlang/OTP ${{matrix.erlang.otp}} / rebar3 ${{matrix.erlang.rebar3}}
    strategy:
      matrix:
        erlang:
          - otp: "17"
            rebar3: "3.10.0"

>             os: ubuntu-18.04

Reviewer 3: ubuntu-18.04 is no longer available.

Author: ok, I'll remove OTP17-20 from Readme.

>           - otp: "17"
>             rebar3: "3.10.0"
>             os: ubuntu-18.04
>           - otp: "18"
>             rebar3: "3.11.1"
>             os: ubuntu-18.04
>           - otp: "19"
>             rebar3: "3.15.2"
>             os: ubuntu-18.04
>           - otp: "20.3.8.26"
>             rebar3: "3.15.2"
>             os: ubuntu-18.04

Author: ```suggestion
```

          - otp: "21.3.8.17"
            rebar3: "3.15.2"
            os: ubuntu-20.04
          - otp: "22.3.4.9"
            rebar3: "3.16.1"
            os: ubuntu-20.04
          - otp: "23.3.4.5"
            rebar3: "3.16.1"
            os: ubuntu-22.04
          - otp: "24.3.4.17"
            rebar3: "3.16.1"
            os: ubuntu-22.04
          - otp: "25.3.2.15"
            rebar3: "3.22.1"
            os: ubuntu-24.04
          - otp: "26.2.5.5"
            rebar3: "3.22.1"
            os: ubuntu-24.04
          - otp: "27.1.2"
            rebar3: "3.22.1"
            os: ubuntu-latest
    steps:

>       - uses: actions/checkout@v2

Reviewer 4: ```suggestion
      - uses: actions/checkout@v4
```
